packages/drivers/odsp-driver/src/createFile.ts

```diff
@@ -165,7 +167,7 @@ export function convertSummaryToSnapshotTreeForCreateNew(summary: ISummaryTree):
 
         const entry: SnapshotTreeEntry = {
             path: encodeURIComponent(key),
-            type: getGitType(summaryObject),
+            type: getGitType(summaryObject) as "blob" | "tree" | "blob",
```

Comment:
getGitType returns blob tree or attachment, but you are casting away attachment here. you do a similar cast in a couple places. i don't quite understand why it's ok for these types not to match

Reply:
Casting was here as an attempt to allow server & client changes to go together without need to have multiple commits, but that did not work out.
That said, it was wrong - getGitType() should never return "attachment", it should return "blob" for it same way as extra copy of that API in r11s was doing it - move that to protocol-base and changes this place and another place where we were special-casing "attachment" case.